de_web_plugin.cpp

```diff
@@ -6082,16 +6101,28 @@ void DeRestPluginPrivate::addSensorNode(const deCONZ::Node *node, const deCONZ::
             fpTuyaSensor.deviceId = i->deviceId();
             fpTuyaSensor.profileId = i->profileId();
 
-            sensor = getSensorNodeForFingerPrint(node->address().ext(), fpTuyaSensor, "ZHATuya");
-            if (!sensor || sensor->deletedState() != Sensor::StateNormal)
-            {
-                addSensorNode(node, fpTuyaSensor, "ZHATuya", modelId, manufacturer);
-            }
-            else
+            fpTuyaSensor.inClusters.push_back(TEMPERATURE_MEASUREMENT_CLUSTER_ID);
+            fpTuyaSensor.inClusters.push_back(RELATIVE_HUMIDITY_CLUSTER_ID);
+            fpTuyaSensor.inClusters.push_back(IAS_ZONE_CLUSTER_ID);
+
+            //So create 3 sensors for this one
+            const QStringList sensorlist = {"ZHATemperature", "ZHAHumidity", "ZHAAlarm" };
+
+            for (int l = 0; l < sensorlist.size(); l++)
             {
-                checkSensorNodeReachable(sensor);
-                checkIasEnrollmentStatus(sensor);
+
+                sensor = getSensorNodeForFingerPrint(node->address().ext(), fpTuyaSensor, sensorlist[l]);
+                if (!sensor || sensor->deletedState() != Sensor::StateNormal)
+                {
+                    addSensorNode(node, fpTuyaSensor, sensorlist[l], modelId, manufacturer);
+                }
+                else
+                {
+                    checkSensorNodeReachable(sensor);
+                    checkIasEnrollmentStatus(sensor);
+                }
             }
```

Comment:
I think this can be a problem since all 3 sensors will have the same finger print.
I'd suggest to just let the existing "fingerprint mechanism" create the sensors.

```cpp
    if (manufacturer.endsWith(QLatin1String("0yu2xgi"))
    {
        fpTemperatureSensor.inClusters.push_back(TUYA_CLUSTER_ID);
        fpTemperatureSensor.inClusters.push_back(TEMPERATURE_MEASUREMENT_CLUSTER_ID);
        fpHumiditySensor.inClusters.push_back(TUYA_CLUSTER_ID);
        fpHumiditySensor.inClusters.push_back(RELATIVE_HUMIDITY_CLUSTER_ID);
        fpAlarmSensor.inClusters.push_back(TUYA_CLUSTER_ID);
        fpAlarmSensor.inClusters.push_back(IAS_ZONE_CLUSTER_ID);
    }
```

No need for the extra `fpTuyaSensor` code since the sensor creation via fpTemperatureSensor, fpHumiditySensor and fpAlarmSensor already exist, or do I miss something?

Reply:
No, I think it s fine like that too.